tests/Integration/Behaviour/Features/Context/Domain/CatalogPriceRuleContext.php

```diff
@@ -0,0 +1,354 @@
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\QueryResult\CatalogPriceRuleForListing;
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\QueryResult\CatalogPriceRuleList;
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\QueryResult\EditableCatalogPriceRule;
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\ValueObject\CatalogPriceRuleId;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Reduction;
+use PrestaShop\PrestaShop\Core\Util\DateTime\DateTime as DateTimeUtil;
+use PrestaShopException;
+use RuntimeException;
+use Shop;
+use SpecificPriceRule;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
+
+class CatalogPriceRuleContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I add catalog price rule :catalogPriceRuleReference with following details:
+     *
+     * @param string $catalogPriceRuleReference
+     * @param TableNode $tableNode
+     */
+    public function addCatalogPriceRule(string $catalogPriceRuleReference, TableNode $tableNode): void
+    {
+        try {
+            $command = $this->createAddCatalogPriceRuleCommand($tableNode);
+
+            /** @var CatalogPriceRuleId $catalogPriceRuleId */
+            $catalogPriceRuleId = $this->getCommandBus()->handle($command);
+
+            $this->getSharedStorage()->set($catalogPriceRuleReference, $catalogPriceRuleId->getValue());
+        } catch (CatalogPriceRuleException|DomainConstraintException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Then catalog price rule :catalogPriceRuleReference should have following details:
+     *
+     * @param string $catalogPriceRuleReference
+     * @param EditableCatalogPriceRule $expectedItem
+     *
+     * @see transformCatalogPriceRule
+     */
+    public function assertCatalogPriceRule(string $catalogPriceRuleReference, EditableCatalogPriceRule $expectedItem): void
+    {
+        $catalogPriceRuleId = $this->getSharedStorage()->get($catalogPriceRuleReference);
+        /** @var EditableCatalogPriceRule $actualItem */
+        $actualItem = $this->getQueryBus()->handle(new GetCatalogPriceRuleForEditing($catalogPriceRuleId));
+        $propertyAccessor = PropertyAccess::createPropertyAccessor();
+
+        $scalarPropertyNames = ['name', 'shopId', 'currencyId', 'countryId', 'groupId', 'fromQuantity'];
+
+        foreach ($scalarPropertyNames as $propertyName) {
+            Assert::assertSame(
+                $propertyAccessor->getValue($expectedItem, $propertyName),
+                $propertyAccessor->getValue($actualItem, $propertyName),
+                sprintf('Unexpected catalogPriceRuleForListing "%s"', $propertyName)
+            );
+        }
+
+        /* isTaxIncluded can't be accessed by property accessor */
```

Comment:
I think if you use `taxIncluded` as the property name it should work 🤔

Reply:
Yeah it seems you are right